mobile-app/app/components/BottomSheetModal.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactElement, useCallback, useRef, useEffect } from 'react'
+import { tailwind } from '@tailwind'
+import { TouchableOpacity, View, Platform } from 'react-native'
+import { BottomSheetModal as Modal, BottomSheetModalProps, useBottomSheetModal } from '@gorhom/bottom-sheet'
+import { useThemeContext } from '@contexts/ThemeProvider'
+import { ThemedProps } from './themed'
+
+type Props = ThemedProps & BottomSheetModalProps & {
+  triggerComponent: ReactElement
+  children: ReactElement
+}
+
+export const BottomSheetModal = (props: Props): JSX.Element => {
+  const bottomSheetModalRef = useRef<Modal>(null)
+  const { dismiss } = useBottomSheetModal()
+  const { isLight } = useThemeContext()
+  const {
+    name,
+    style,
+    children,
+    triggerComponent,
+    snapPoints = ['100%', '50%'],
+    light = tailwind('bg-gray-100 text-black'),
+    dark = tailwind('bg-gray-600 text-white text-opacity-90')
+  } = props
+
+  const openModal = useCallback(() => {
+    bottomSheetModalRef.current?.present()
+  }, [])
+
+  useEffect(() => {
+    return () => {
+      if (Platform.OS !== 'web') {
+        dismiss(name)
+      }
+    }
+  }, [])
+
+  if (Platform.OS === 'web') {
+    return <></>
+  }
+
+  return (
+    <View>
+      <TouchableOpacity onPress={openModal}>
+        {triggerComponent}
+      </TouchableOpacity>
+      <Modal
```

Comment:
From figma, jonathan uses a close button, is it possible to change the default handle using [handleComponent](https://gorhom.github.io/react-native-bottom-sheet/props#handlecomponent) ?

Reply:
In the design we are using a modal but in actual implementation we are using bottom sheet, which will get close on swipe down the modal, @thedoublejay  let me know if we need to add the close button here